Source/Core/Core/DSP/Interpreter/DSPInterpreter.cpp

```diff
@@ -394,12 +394,16 @@ s16 Interpreter::GetAXHigh(s32 reg) const
 s64 Interpreter::GetLongAcc(s32 reg) const
 {
   const auto& state = m_dsp_core.DSPState();
-  return static_cast<s64>(state.r.ac[reg].val << 24) >> 24;
+  return static_cast<s64>(state.r.ac[reg].val);
 }
 
 void Interpreter::SetLongAcc(s32 reg, s64 value)
 {
   auto& state = m_dsp_core.DSPState();
+  if (value & 0x0080'0000'0000)
+    value |= 0xffff'ff00'0000'0000;
+  else
+    value &= 0x00ff'ffff'ffff;
```

Comment:
```suggestion
    value &= 0x0000'00ff'ffff'ffff;
```

To be consistent length-wise with the branch above it